lib/promise.js

```diff
@@ -47,6 +50,13 @@ for (var method in Promise) {
 
 var bluebirdThen = Promise.prototype._then;
 Promise.prototype._then = function (didFulfill, didReject, didProgress, receiver, internalData) {
+  if (SequelizePromise.prototype.sequelize.options.namespace) {
```

Comment:
definitely not `context`, since that has a different meaning in CLS :). But simply `cls` might be good, especially if we decide to use a boolean flag instead